x-pack/plugins/uptime/public/lib/helper/component_test_helpers.tsx

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { ReactElement } from 'react';
+import { render as reactRender } from '@testing-library/react';
+import { Router } from 'react-router-dom';
+import { MemoryHistory } from 'history/createMemoryHistory';
+import { createMemoryHistory, History } from 'history';
+import { I18nProvider } from '@kbn/i18n/react';
+import { mountWithIntl, renderWithIntl, shallowWithIntl } from '@kbn/test/jest';
+import { coreMock } from 'src/core/public/mocks';
+import {
+  KibanaContextProvider,
+  KibanaServices,
+} from '../../../../../../src/plugins/kibana_react/public';
+import { MountWithReduxProvider } from './helper_with_redux';
+import { AppState } from '../../state';
+
+interface KibanaProps {
+  services?: KibanaServices;
+}
+
+interface KibanaProviderOptions {
+  coreOptions?: any;
+  kibanaProps?: KibanaProps;
+}
+
+interface MockKibanaProviderProps extends KibanaProviderOptions {
+  children: ReactElement;
+}
+
+interface MockRouterProps extends MockKibanaProviderProps {
+  history?: History;
+}
+
+interface RenderRouterOptions extends KibanaProviderOptions {
+  history?: History;
+  renderOptions?: any;
+  state?: any;
```

Comment:
```suggestion
  state?: AppState;
```

Reply:
The one thing I don't love about requiring the full app state, is that you have to provide pieces of state that are unnecessary to the particular component you are testing in order to satisfy the interface.

Reply:
Maybe we could use generics for this and specify the smaller slice of state you need.

Reply:
Good point; can we use `Partial<AppState>` instead, and when we

```typescript
<MountWithReduxProvider store={store}>
```

change it to:

```typescript
<MountWithReduxProvider store={...mockStore, ...userStore}>
```